driver-async/src/test/unit/com/mongodb/async/client/FindIterableSpecification.groovy

```diff
@@ -76,7 +77,7 @@ class FindIterableSpecification extends Specification {
                 .partial(false)
                 .collation(null)
                 .comment('my comment')
-                .hint(new Document('hint', 1))
+                .hintString('a_1')
```

Comment:
Unfortunately, I think we need three tests :(

1. hint
2. hintString
3. hint + hintString

Maybe a separate test method so that it's not so verbose.

Reply:
We are testing both: hintString and hint + hintString not sure we need another.

Reply:
The latter would be to show which gets precedence when both are set.  I'll leave it to your discretion whether to include a test for that.